controllers/drpolicy.go

```diff
@@ -38,15 +40,20 @@ func drClusterSecretsDeploy(
 	drclusters *rmn.DRClusterList,
 	secretsUtil *util.SecretsUtil,
 	rmnCfg *rmn.RamenConfig,
+	log logr.Logger,
 ) error {
 	if !rmnCfg.DrClusterOperator.DeploymentAutomationEnabled ||
 		!rmnCfg.DrClusterOperator.S3SecretDistributionEnabled {
 		return nil
 	}
 
-	drPolicySecrets, err := drPolicySecretNames(drpolicy, drclusters, rmnCfg)
+	drPolicySecrets, err := drPolicySecretNames(drpolicy, drclusters, rmnCfg, log)
 	if err != nil {
-		return err
+		// For cluster deploy, it is ok to deploy only what available so far.
+		// Fail it only if no secrets are available.
+		if len(drPolicySecrets) == 0 {
```

Comment:
This is needed only for deployment — no need to have the same for the undeployment. We need the undeployment to fail in order to not leak resources.